### DISCUSSION TITLE
Security Onion 2.3.110 & 120 - Osquery versions not updating.

### DISCUSSION THREAD

**Discussion by rsharris3** [Security-Onion-Solutions/securityonion#8026] [Unsupported Versions] [answered]
We have a distributed environment that has been running and upgraded multiple versions from early 2.3 without issues.  We added some additional forward sensor nodes while we were on version 2.3.110 and noticed that the Osquery version on these new nodes are 4.5.1 and do not appear to be updating automatically to 5.1.0 as the rest of the environment are at.  We used soup to upgrade the environment to 2.3.120 in hopes that this would initiate the Osquery update, but that did not happen.  The environment was successfully upgraded and is working as expected, except the handful of recently installed sensors are at the older version of Osquery.

Am I correct that these should update automatically?  Can someone point me in the direction of the logs or reason that may help identify why they aren't updating?  

Thanks!

**Comment by defensivedepth** [comment]
@rsharris3 Has this been resolved?

**Reply by rsharris3** [reply]
@defensivedepth,  not yet.  I have a troubleshooting session setup for this morning.  Will keep you posted on the outcome.

Thanks.

**Reply by rsharris3** [reply]
@defensivedepth,  during this mornings session, we removed the /usr/local/so-launcher/bin/osqueryd-updates/ folder as suggested.  Since then the agents still have not been updated and the folder and subfolders have not been recreated.  The agents are still showing online and seem to checking in about every hour to the fleet dm portal.

Let me know if you have other suggestions.  Thanks!

**Reply by tiger-greenhell** [reply]
Hi.

We have a similar problem on version 2.3.120.
Have you solved this problem?

**Comment by rsharris3** [comment, marked as answer]
Was able to get a resolution working with support on this.  

Suggestion from support from the Manager ran:

salt \* cmd.run "tar -xf /var/so-launcher/securityonion/osqueryd-staging/linux/osqueryd-stable.tar.gz -C /usr/local/so-launcher/bin && systemctl restart launcher.so-launcher.service"

This will use the original binary in the tar.gz and replace what appears to be a corrupted one. It will manually extract and replace it, then restart the service. 

You should see the nodes updated in the console once FleetDM fetches new information about the host. If you manually fetch the agent in FleetDM do 'Refetch' on the host. You can check to make sure it updates.

I also had one node that this failed on, got this error when running the command.
    gzip: stdin: unexpected end of file
    tar: Unexpected EOF in archive
    tar: Unexpected EOF in archive
    tar: Error is not recoverable: exiting now

For this node, I replaced the osqueryd-stable.tar.gz in the /var/so-launcher/securityonion/osqueryd-staging/linux/ folder with one that worked on the manager node.  The files were different sizes, so they were definitely different.  I then ran the command directly on that node which worked and updated osquery.

Thanks!
